pkg/codec.go

```diff
@@ -278,7 +282,7 @@ func (m *Hessian2Codec) Decode(ctx context.Context, message remote.Message, in r
 func (m *Hessian2Codec) decodeHeartbeatBody(ctx context.Context, header *dubbo.DubboHeader, message remote.Message, in remote.ByteBuffer) error {
 	// for heartbeat, there is no need to decode the body
 
-	// todo(DMwangnima): process heartbeat with UnknownHandler
+	message.SetMessageType(remote.Heartbeat)
```

Comment:
Do we need to consume the last Null?

Reply:
dubbo-java use
```java
// in hessian2, this means the body is 'N'
Arrays.equals(payload, getNullBytesOf(getSerializationById(proto)))
```
to determine if this response is a heartbeat when IsEvent flag in dubbo header has been set.
As a result, we have to consume the entire body.